pandas/core/frame.py

```diff
@@ -1978,7 +1978,48 @@ def memory_usage(self, index=True, deep=False):
         return result
 
     def transpose(self, *args, **kwargs):
-        """Transpose index and columns"""
+        """
+        Transpose index and columns.
+
+        Reflect the dataframe over its main diagonal by writing rows as columns
+        and vice-versa.
+
+        Parameters
+        ----------
+        *args
+            These arguments will be passed to
+            :func:`pandas.compat.numpy.function.validate_transpose`.
+        **kwargs
+            Keyword arguments used to create the transposed DataFrame.
+
+        Returns
+        -------
+        DataFrame
+            The transposed DataFrame.
+
+        See Also
+        --------
+        numpy.transpose : Permute the dimensions of a given array.
+
+        Examples
+        --------
+        >>> d = {'col1': [1, 2], 'col2': [3, 4]}
+        >>> df = pd.DataFrame(data=d)
+        >>> df
+           col1  col2
+        0     1     3
+        1     2     4
+
+        >>> df.T
+              0  1
+        col1  1  2
+        col2  3  4
+
+        >>> df.transpose()
+              0  1
+        col1  1  2
+        col2  3  4
+        """
```

Comment:
Fantastic to see the usage of both T and transpose!
If possible, please add an example with a non-square DataFrame, e.g. 2 rows and 5 columns to show off some functionality - or maybe add some information about the fact it can be applied to square and non-square DataFrames?

Reply:
Done. How do you like it now? :)